### PR TITLE
[INLONG-8643][Sort] Support Iceberg source

### LINKED ISSUE #8643: [Feature][Sort] Add Iceberg source on flink 1.15
### Description

_No response_

### Use case

_No response_

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by vernedeng**
- Fixes #8643 

### Motivation

Support iceberg source of flink version 1.15 and iceberg version 1.3.1

### Modifications

Add iceberg source

### Verifying this change

*(Please pick either of the following options)*

- [ ] This change is a trivial rework/code cleanup without any test coverage.

- [ ] This change is already covered by existing tests, such as:
  *(please describe tests)*

- [ ] This change added tests and can be verified as follows:

  *(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### Documentation

  - Does this pull request introduce a new feature? (yes / no)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)
  - If a feature is not applicable for documentation, explain why?
  - If a feature is not documented yet in this PR, please create a follow-up issue for adding the documentation

### CODE REVIEW

File: inlong-sort/sort-core/src/test/java/org/apache/inlong/sort/parser/Iceberg2StarRocksSqlParserTest.java [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+
+    private String groupId = "b_test_wk_0801";
+    private String streamId = "b_test_wkstream_0801";
+
+    // iceberg
+    private String uri = "thrift://ss-qe-nginx-tauth.tencent-distribute.com:9991";
+    private String icDatabase = "test_inlong_iceberg";
+    private String icTable = "t_ic_gwe0000000260d";
+    private String catalogName = "HIVE";
+    private String warehouse = "hdfs://qy-teg-2-v3/user/tdw/warehouse/test_inlong_iceberg.db/t_ic_gwe0000000260d";
```
Comment (EMsnap) [inline inlong-sort/sort-core/src/test/java/org/apache/inlong/sort/parser/Iceberg2StarRocksSqlParserTest.java:59]: please remove online address

**Comment by EMsnap** [comment]
Maybe we don't need to modify iceberg source code since we are not implementing audit metrics yet.

**Comment by dockerzhang** [comment]
@vernedeng please add license for copyed files.

**Comment by vernedeng** [comment]
> Maybe we don't need to modify iceberg source code since we are not implementing audit metrics yet.

OK

**Review by yunqingmoswu — APPROVED** [review]

**Comment by vernedeng** [comment]
> @vernedeng please add license for copyed files.

Flxed

**Review by dockerzhang — APPROVED** [review]

**Review by EMsnap — APPROVED** [review]

**Pull request merged**